wholegames-mocking.Rmd

```diff
@@ -0,0 +1,105 @@
+# vcr and httptest {#mocking-pkgs-comparison}
+
+We have just followed very similar processes to add HTTP testing infrastructure involving mock files to exemplighratia
+
+* Adding a package as a Suggests dependency;
+* Creating a helper file that in particular loads this package before each test;
+* Tweaking tests, in some cases wrapping our tests into functions that allows to record API responses in mock files and to play them back from said mock files; in other cases (only with httptest), creating mock files ourselves.
+
+Now, there were a few differences. 
+We won't end up advocating for one package in particular since both have their merits, but we do hope to help you differentiate the two packages.
+
+## Setting up the infrastructure
+
+To set up the HTTP testing infrastructure, in one case you need to run `vcr::use_vcr()` and in another case you need to run `httptest::use_httptest()`. Not too hard to remember.
+
+## Calling mock files
+
+As mentioned before, vcr and httptest both use mock files but they call them differently.
+
+In vcr they are called both **fixtures** and **cassettes**.
+In httptest they are called **mock files**.
+Note that fixtures is not as specific as cassettes and mock files: cassettes and mock files are fixtures, but anything (a csv file of input for instance) you use to consistently test your package is a fixture.
+
+## Naming mock files
+
+With vcr the `use_cassette()` call needs to include a name that will be used to create the filename of the mock file.
+The help of `?use_cassette` explains some criteria for naming them, such as the fact that cassette names need to be unique.
+Now if you wrap your whole `test_that()` block in them you might just as well use a name similar to the test name, and you already make those meaningful, right?
+
+With httptest the mock filepaths are translated from requests according to several rules that incorporate the request method, URL, query parameters, and body. 
+If you use `with_mock_dir()` you need a name for the directory under which the mock files are saved, and you can make it meaningful.
+
+Also note that with vcr one file contains several HTTP interactions (requests and responses) whereas with httptest one file contains one response only (and the filename helps matching it to a request).
```

Comment:
I'd clarify to say that one vcr fixture CAN contain more than one http interaction, doesn't always have >1